java/apirag-core/src/main/java/com/datasqrl/ai/models/AbstractModelConfiguration.java

```diff
@@ -0,0 +1,65 @@
+package com.datasqrl.ai.models;
+
+import com.datasqrl.ai.util.ErrorHandling;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.MapConfiguration;
+
+@AllArgsConstructor
+public abstract class AbstractModelConfiguration implements ModelConfiguration {
+
+  public static final String MODEL_NAME_KEY = "name";
+  public static final String MAX_INPUT_TOKENS_KEY = "max_input_tokens";
+  public static final String MAX_OUTPUT_TOKENS_KEY = "max_output_tokens";
+  public static final String TEMPERATURE_KEY = "temperature";
+  public static final double TEMPERATURE_DEFAULT = 0.7;
+  public static final String TOP_P_KEY = "top-p";
```

Comment:
@mbroecheler Shall we try to stick to snake_case for the configuration? Most LLMs are configured so. So we might want to unify our configs to use underscores throughout. Wdyt? `top_p`

Reply:
Yes, good eye!